pip/wheel.py

```diff
@@ -379,6 +379,40 @@ def is_entrypoint_wrapper(name):
 
     maker = ScriptMaker(None, scheme['scripts'])
 
+    def get_padded_executable():
+        executable = sys.executable
+        if ' ' in executable:
+            needs_padding = True
+        else:
+            platform = sys.platform
+            if platform.startswith('linux'):
+                platform = 'linux'
+            # See http://www.in-ulm.de/~mascheck/various/shebang/#length
+            shebang_max_lengths = {
+                'linux': 127,
+                'darwin': 512,
+            }
```

Comment:
I couldn't see the limits you are using here confirmed in the URL you mention, and there are a lot of other different limits mentioned there. It seems to me that there will be plenty of systems that could still have the issue even with this patch. I can't comment on how obscure those systems might be, of course.